File: Chapter 05/C Code/5_5.c
#include <stdio.h>
#include <stdbool.h>

bool is_prime(int num) {
    if (num < 2) {
        return false;
    }
    for (int i = 2; i * i <= num; i++) {
        if (num % i == 0) {
            return false;
        }
    }
    return true;
}

int main() {
    int L2[] = {2, 3, 4, 5, 6, 7, 8, 9, 10};
    int n = sizeof(L2) / sizeof(L2[0]);
    int L1[n];
    int count = 0;
    for (int i = 0; i < n; i++) {
        if (is_prime(L2[i])) {
            L1[count++] = L2[i];
        }
    }
    printf("L1 = { ");
    for (int i = 0; i < count; i++) {
        printf("%d ", L1[i]);
    }
    printf("}\n");
    return 0;
}

File: Chapter 05/Python code/5_5.py

L2 = [45,31,17,33,97,118]
L1 = []

for n in L2:
    is_prime = True    
    for i in range(2, n):
        if n % i == 0:
            is_prime = False
            break
            
    if is_prime == True:
        L1.append(n)

print("L1:", L1)

